storage/allocator_test.go

```diff
@@ -627,6 +627,245 @@ func TestAllocatorRemoveTarget(t *testing.T) {
 	}
 }
 
+func TestAllocatorComputeAction(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	stopper, _, _, a := createTestAllocator()
+	defer stopper.Stop()
+
+	// Each test case should describe a repair situation which has a lower
+	// priority than the previous test case.
+	testCases := []struct {
+		zone           config.ZoneConfig
+		desc           proto.RangeDescriptor
+		expectedAction allocatorAction
+	}{
+		// Needs Three replicas, have two
```

Comment:
Test 1 - 2/3 -- aaAdd
Test 2 - 4/5 -- aaAdd
Test 3 - 4/3 -- aaRemove
Test 4 - 5/3 -- aaRemove
Test 5 - 3/3 -- aaNoop

Can you add tests for aaRemoveDead and aaRepairButNoQuorum (when you add it) ?